reference/enterprise-api-ref/inventory.markdown

```diff
@@ -0,0 +1,213 @@
+---
+layout: default
+title: Inventory API
+published: true
+tags: [reference, enterprise, API, reporting, URI]
+---
+Inventory API allows to access inventory reports and variable dictionary.
+
+## Inventory Reports
+   
+
+**URI:** https://hub.cfengine.com/api/inventory
+
+**Method:** POST
+
+
+**Parameters:**
+
+* **select** *(array)*
+    Fields for selecting. Required parameter.
+    List of fields name you can obtain through [List of variables][Inventory API#List of variables] call described below.
+* **filter** *(json object)*
+    Filter data. Optional parameter.
+* **sort** *(string)*
+    Field name for sorting with "-" for DESC order. Optional parameter.
+* **skip** *(integer)*
+    Number of results to skip for the processed
+    query. Optional parameter.
+* **limit**  *(integer)*
+    Limit the number of results in the query. Default value is 50, max value is 1000.
+* **hostContextExclude** *(array)*
+    Includes only results that concern hosts which have all specified CFEngine contexts (class) set. Optional parameter.
+* **hostContextInclude** *(array)*
+    Excludes results that concern hosts which have specified CFEngine context (class) set. Hosts that have at least one of the specified contexts set will be excluded from the results. Optional parameter.
+
+**Example Request Body:**
+
+
+```
+{  
+   "sort":"Host name",
+   "filter":{  
+      "Host name":{  
+         "l":"%value1%",
+         "nil":"%value2%"
+      }
+   },
+   "select":[  
+      "Host name",
+      "OS type",
+      "IPv4 addresses",
+      "CFEngine version",
+      "Ports listening"
+   ],
+   "hostContextExclude":[  
+      "policy_server"
+   ]
+}
+```
+
+**Example response:**
+
+
+```
+{
+  "data": [
+    {
+      "header": [
+        {
+          "columnName": "Host name",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "OS type",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "IPv4 addresses",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "CFEngine version",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "Ports listening",
+          "columnType": "STRING"
+        }
+      ],
+      "query": "SELECT  values->>'Host name' AS \"Host name\",  values->>'OS type' AS \"OS type\",  cf_clearSlist(values->>'IPv4 addresses') AS \"IPv4 addresses\",  values->>'CFEngine version' AS \"CFEngine version\",  cf_clearSlist(values->>'Ports listening') AS \"Ports listening\" FROM inventory_new WHERE values->>'Host name' LIKE '%mender%' AND values->>'Host name' NOT ILIKE '%io%' ORDER BY values->>'Host name' ASC ",
+      "queryTimeMs": 30,
+      "rowCount": 2,
+      "rows": [
+        [
+          host1.cfengine.com",
+          "linux",
+          "128.30.23.33",
+          "3.10.0",
+          "22, 25, 80, 443, 5308, 9000"
+        ],
```

Comment:
what if a host does not have a specific column, how would that response look? (maybe include in example )